runtime/src/accounts_db.rs

```diff
@@ -1351,6 +1387,50 @@ impl AccountsDB {
         hasher.result()
     }
 
+    pub fn blake3_hash_account_data(
+        slot: Slot,
+        lamports: u64,
+        owner: &Pubkey,
+        executable: bool,
+        rent_epoch: Epoch,
+        data: &[u8],
+        pubkey: &Pubkey,
+    ) -> Hash {
+        if lamports == 0 {
+            return Hash::default();
+        }
+
+        let mut hasher = blake3::Hasher::new();
+
+        hasher.update(&lamports.to_le_bytes());
+
+        hasher.update(&slot.to_le_bytes());
+
+        hasher.update(&rent_epoch.to_le_bytes());
+
+        hasher.update(&data);
+
+        if executable {
+            hasher.update(&[1u8; 1]);
+        } else {
+            hasher.update(&[0u8; 1]);
+        }
+
+        hasher.update(&owner.as_ref());
+        hasher.update(&pubkey.as_ref());
+
+        Hash(<[u8; solana_sdk::hash::HASH_BYTES]>::try_from(hasher.finalize().as_slice()).unwrap())
+    }
+
+    fn get_blake3_slot(operating_mode: &OperatingMode) -> Slot {
+        match operating_mode {
+            OperatingMode::Development => 0,
```

Comment:
@carllin Well, this makes devnet snapshot not loadable. Obviously, current snapshots are hashed with sha256, not blake3 yet..